api/src/main/java/io/opentelemetry/trace/attributes/BooleanAttributeSetter.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.trace.attributes;
+
+import io.opentelemetry.trace.Span;
+import javax.annotation.concurrent.Immutable;
+
+/** Defines the behavior for a span attribute with boolean values. */
+@Immutable
+public class BooleanAttributeSetter {
```

Comment:
Consider to use autovalue.

Reply:
Why?

Reply:
To avoid checking for null the key and have autogenerated hash/equal if we need them :)

Reply:
Not a strong opinion, as I said is a consideration.